src/telemetry/writer.cpp

```diff
@@ -0,0 +1,233 @@
+/*
+ * Author: Robertas Norkus
+ * Organisation: HYPED
+ * Date: January 2020
+ * Description:
+ *
+ *    Copyright 2019 HYPED
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <string>
+
+#include "writer.hpp"
+#include "data/data.hpp"
+
+namespace hyped {
+namespace telemetry {
+
+  // additional data points that are displayed in the GUI data section
+  // FEEL FREE TO EDIT. More info: https://github.com/Hyp-ed/hyped-2020/wiki/Adding-new-data-points
+  void Writer::packAdditionalData()
+  {
+    rjwriter_.Key("additional_data");
+    rjwriter_.StartArray();
```

Comment:
There `StartArray()` and `EndArray()` functions seem always paired. Can you think about some scoped context that would ensure the correct pairing. I am thinking of something similar to `utils::concurrent::ScopedLock`.

Reply:
I do not control rjwriter_, as it is rapidjson object.